vault/vault.go

```diff
@@ -1,11 +1,45 @@
 package vault
 
 import (
+	"fmt"
+	"strings"
 	"sync"
 
 	"github.com/hashicorp/vault/api"
 )
 
+// EngineType is an identifier for an engine
+type EngineType string
+
+// EngineTypeKeyValueV1 is the identifier for version 1 of the key/value engine.
+const EngineTypeKeyValueV1 EngineType = "kv"
+
+// EngineTypeKeyValueV2 is the identifier for version 2 of the key/value engine.
+const EngineTypeKeyValueV2 EngineType = "kv-v2"
```

Comment:
I think I'd group these together in a single `const` block for clarity.

Reply:
Good call.  I did the same for the AuthType consts that were just below as well...